corehq/apps/hqadmin/management/commands/static_analysis.py

```diff
@@ -81,6 +82,24 @@ def show_custom_modules(self):
         self.logger.log("commcare.static_analysis.custom_module_count", custom_module_count)
         self.logger.log("commcare.static_analysis.custom_domain_count", custom_domain_count)
 
+    def show_js_dependencies(self):
+        proc = subprocess.Popen(["./scripts/codechecks/hqDefine.sh", "static-analysis"], stdout=subprocess.PIPE)
+        output = proc.communicate()[0].strip().decode("utf-8")
+        (hqdefine_todo, hqdefine_done, requirejs_todo, requirejs_done) = output.split(" ")
```

Comment:
There's also a `./scripts/static-analysis.sh` file for running cmdline tools.  I added that in the hopes it'd be an easier way to do that than calling them from python code.  However, in practice I've found extracting the information I need kind of a pain due to my lack of familiarity with `sed` and friends.  I'm curious whether you tried out that approach, and what your thoughts are - would it be better to just do everything in this management command?

Reply:
My bash scripting is poor, so I'd rather do things in python. Someone else (Danny?) originally wrote the `hqDefine` script - I'd have done the whole thing in python, with as many `subprocess` calls as it took.